From edd6d6e2805894e21afe51276d1af65d5395180d Mon Sep 17 00:00:00 2001
From: Andy Wingo <wingo@pobox.com>
Date: Sun, 07 Aug 2016 10:23:50 +0000
Subject: Snarfing .x depends on built headers

* libguile/Makefile.am (BUILT_INCLUDES): New variable.
  (BUILT_SOURCES): Put .i and other generated .h to BUILT_INCLUDES.
  (.c.x, .c.doc): Depend on BUILT_INCLUDES.  Fixes
  https://bugs.gentoo.org/show_bug.cgi?id=590528.
---
diff --git a/libguile/Makefile.am b/libguile/Makefile.am
index cd85604..4c54744 100644
--- a/libguile/Makefile.am
+++ b/libguile/Makefile.am
@@ -430,9 +430,9 @@ DOT_I_FILES = vm-i-system.i vm-i-scheme.i vm-i-loader.i
 .c.i:
 	$(AM_V_GEN)$(GREP) '^VM_DEFINE' $< > $@
 
-BUILT_SOURCES = cpp-E.c cpp-SIG.c libpath.h \
-    scmconfig.h \
-    $(DOT_I_FILES) $(DOT_X_FILES) $(EXTRA_DOT_X_FILES)
+BUILT_INCLUDES = $(DOT_I_FILES) scmconfig.h
+BUILT_SOURCES = cpp-E.c cpp-SIG.c libpath.h $(BUILT_INCLUDES) \
+    $(DOT_X_FILES) $(EXTRA_DOT_X_FILES)
 
 # Force the generation of `guile-procedures.texi' because the top-level
 # Makefile expects it to be built.
@@ -730,9 +730,9 @@ snarfcppopts = $(DEFS) $(AM_CPPFLAGS) $(CPPFLAGS) $(CFLAGS)
 
 SUFFIXES = .x .doc
 
-.c.x:
+.c.x: $(BUILT_INCLUDES)
 	$(AM_V_SNARF)$(GUILE_SNARF) -o $@ $< $(snarfcppopts)
-.c.doc:
+.c.doc: $(BUILT_INCLUDES)
 	$(AM_V_SNARF)./guile-snarf-docs -o $@ $< -- $(snarfcppopts)
 
 $(DOT_X_FILES) $(EXTRA_DOT_X_FILES): scmconfig.h snarf.h guile-snarf.in version.h
--
cgit v0.9.0.2
